code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/EdgePersonalizationResponseHandler.java

```diff
@@ -143,6 +144,13 @@ class EdgePersonalizationResponseHandler {
                 if (inAppRules != null) {
                     final List<LaunchRule> rulesToReplace = new ArrayList<>();
                     for (final Map.Entry<Surface, List<LaunchRule>> entry : inAppRules.entrySet()) {
+                        // MOB-21846 - iam items are returned in reverse priority order, so we need
+                        // to
+                        // flip the order of the list prior to saving them and hydrating the rules
+                        // engine. this allows the highest priority item to be shown first when
+                        // rules
+                        // engine evaluates top-down
+                        Collections.reverse(entry.getValue());
```

Comment:
nit: can we clean up the comments to remove the unnecessary extra lines?

Reply:
the formatting here is being applied by spotless. i'll play around with line lengths to see if it can have a better look after formatting.